panel/pane.py

```diff
@@ -560,6 +560,74 @@ def _img(self):
         return b.getvalue()
 
 
+def latex_to_img(text, size=25):
+    """
+    Returns PIL image for LaTeX equation text, using matplotlib's rendering.
+    Usage: latex_to_img(r'$\frac(x}{y^2}$')
+    From https://stackoverflow.com/questions/1381741.
+    """
+    import matplotlib.pyplot as plt
+    from PIL import Image, ImageChops
+    import io
+    
+    buf = io.BytesIO()
+    plt.rc('text', usetex=True)
+    plt.rc('font', family='serif')
+    plt.axis('off')
+    plt.text(0.05, 0.5, '{text}'.format(text=text), size=size)
+    plt.savefig(buf, format='png')
+    plt.close()
+
+    im = Image.open(buf)
+    bg = Image.new(im.mode, im.size, (255, 255, 255, 255))
+    diff = ImageChops.difference(im, bg)
+    diff = ImageChops.add(diff, diff, 2.0, -100)
+    bbox = diff.getbbox()
+    return im.crop(bbox)
+
+
+def make_transparent(img, bg=(255, 255, 255, 255)):
+    """Given a PIL image, makes the specified background color transparent."""
+    from PIL import Image
+    img = img.convert("RGBA")
+    clear = bg[0:3]+(0,)
+    pixdata = img.load()
+    
+    width, height = img.size
+    for y in range(height):
+        for x in range(width):
+            if pixdata[x,y] == bg:
+                pixdata[x,y] = clear
+    return img
+
+
+class LaTeX(PNG):
+    """
+    Matplotlib-based LaTeX-syntax equation.
+    Requires matplotlib and pillow.
+    """
+
+    # Precedence is dependent on the data type
+    precedence = None
+
+    size = param.Number(default=25, bounds=(1, 100), doc="""
+        Scales the size of the rendered equation.""")
+
+    @classmethod
+    def applies(cls, obj):
+        if hasattr(obj, '_repr_latex_'):
```

Comment:
Should return False if matplotlib not available.

Reply:
That sounds like a good idea, but I can't see how to make it work.  `pane.Matplotlib` checks `sys.modules` for `matplotlib`, which is valid because there shouldn't be any way to have a Matplotlib figure in memory without `matplotlib` having been imported.  Here, however, the user is just supplying a string or maybe an `IPython.display.Latex` object, and so the user need not have imported anything.  So if we wanted `applies` to return False, wouldn't we need to try actually importing matplotlib and pillow within the `applies` method, so that we know whether they are available?  That doesn't seem feasible; I wouldn't want Panel to have to import those two libraries unless we actually have an instance of `pane.LaTeX` already.  So I don't *think* what you are proposing here is feasible.

Reply:
It's not ideal but importing matplotlib and pillow in the `_repr_latex_` branch is not a major issue unless you expect that we register another pane that handles `_repr_latex_`, because if you're already in that branch it will pick that pane and simply fail loudly later when the pane is actually being rendered.

Reply:
Ok.  I guess the benefit of adding that will be that if an object provides multiple repr methods and matplotlib or pillow is not installed, then display will drop back to one of those other repr methods.